src/server/game/AI/CoreAI/PetAI.cpp

```diff
@@ -420,8 +420,11 @@ void PetAI::DoAttack(Unit* target, bool chase)
 
             if (me->HasUnitState(UNIT_STATE_FOLLOW))
                 me->GetMotionMaster()->Remove(FOLLOW_MOTION_TYPE);
-
-            me->GetMotionMaster()->MoveChase(target, me->GetPetChaseDistance(), float(M_PI));
+            
+            // Pets with ranged attacks should not care about the chase angles or tolerances at all.
+            float chaseDistance = me->GetPetChaseDistance();
+            ChaseAngle angle = ChaseAngle(chaseDistance == 0.f ? float(M_PI) : 0.f, chaseDistance == 0.f ? float(M_PI_4) : float(M_PI * 2));
```

Comment:
I would prefer to see this as a readable if/else block than an unreadable line with 2 ? :